### PR TITLE
Read events data from BigQuery within CorrectnessTest.

### PR DESCRIPTION
**Pull request by yunyeng**
This change is [Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)

### CODE REVIEW

**Review by yunyeng — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @kungfucraig and @wangyaopw)

___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 113 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_R15qH1Q9s0bXzB04d:-M_RBhalA4TXGrLGb407:b-di9kt1) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/621fc444b4cec274c19e2e94339ecfb5c7578fdd/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L113)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

> ```
> README
> ```

readme?

I think these parameter should be read from a config file, i.e., a textproto, so we don't need to re-build the binary each time.

also shouldn't publisherId also be a parameter?
</blockquote></details>

I will be moving these parameters into flag in the next Pull Request. Put some sample parameters to test the query. I will document them as well. Also, added publisher_id as parameter. Thank you.

**Review by wangyaopw — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @kungfucraig, @wangyaopw, and @yunyeng)

___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 204 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_RHkax7ZTEQ48reP8b:-M_RHkax7ZTEQ48reP8c:bnbtazx) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L204)):*
> ```Kotlin
>   ): List<LabeledEvent> {
>     val queryConfig =
>       buildQueryConfig(publisher, sex, ageGroup, socialGrade, complete, startDate, endDate)
> ```

If there are K publishers, eventually we need K sketches. so we need to either
1. do K queries, 1 for each publisher.
2. do 1 query and get a full list, then filter the list in kotlin to get K sublists
Not sure which one is cheaper.

File: src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt [hunk 1]
```diff
@@ -96,8 +106,29 @@ class CorrectnessImpl(
 
   suspend fun process(
     relationalDatabase: KingdomRelationalDatabase,
-    relationalDatabaseTestHelper: DatabaseTestHelper
+    relationalDatabaseTestHelper: DatabaseTestHelper,
+    bigQuery: BigQuery
   ) {
+
+    // TODO(@yunyeng): Move parameters into flags and document them.
```
Comment (kungfucraig) [inline src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt:113]: I'd move this to a test case once you add flags.

**Review by kungfucraig — APPROVED** [review]

**Review by kungfucraig — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)* status: 0 of 5 files reviewed, 4 unresolved discussions (waiting on @wangyaopw and @yunyeng)

___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 291 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_RV1Ty-kIYRqau2CET:-M_RV1Ty-kIYRqau2CEU:b-eca81j) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L291)):*
> ```Kotlin
>       "M" -> LabeledEvent.Sex.MALE
>       "F" -> LabeledEvent.Sex.FEMALE
>       else -> LabeledEvent.Sex.SEX_UNSPECIFIED
> ```

Raise and exception for the else clauses?

**Review by kungfucraig — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)* status: 0 of 5 files reviewed, 5 unresolved discussions (waiting on @wangyaopw and @yunyeng)

___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 289 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_RVmkoDh2p_6kiMqK0:-M_RVmkpDCdgfF7BqhGE:bpbaio6) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L289)):*
> ```Kotlin
>   private fun selectSex(sex: String): LabeledEvent.Sex {
>     return when (sex) {
>       "M" -> LabeledEvent.Sex.MALE
> ```

Use a map of str->enum value to store these. Have these converters driven from the map and drive your command line parsing from map.values().

**Review by yunyeng — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)* status: 0 of 5 files reviewed, 5 unresolved discussions (waiting on @kungfucraig and @wangyaopw)

___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 113 at r1](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_R15qH1Q9s0bXzB04d:-M_RnHeo0wbvTyxxhTuI:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/621fc444b4cec274c19e2e94339ecfb5c7578fdd/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L113)):*
<details><summary><i>Previously, yunyeng (Yunus Yen) wrote…</i></summary><blockquote>

I will be moving these parameters into flag in the next Pull Request. Put some sample parameters to test the query. I will document them as well. Also, added publisher_id as parameter. Thank you.
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 113 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_P4b41AWSSeLSswST6-r2-113:-M_RnHUJDKmTQQ-JhiUw:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L113)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

I'd move this to a test case once you add flags.
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 204 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_RHkax7ZTEQ48reP8b:-M_RnF0qB165ApB8VQNV:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L204)):*
<details><summary><i>Previously, wangyaopw (Yao Wang) wrote…</i></summary><blockquote>

If there are K publishers, eventually we need K sketches. so we need to either
1. do K queries, 1 for each publisher.
2. do 1 query and get a full list, then filter the list in kotlin to get K sublists
Not sure which one is cheaper.
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 289 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_RVmkoDh2p_6kiMqK0:-M_Rn9W3ETF9EM910CuO:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L289)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

Use a map of str->enum value to store these. Have these converters driven from the map and drive your command line parsing from map.values().
</blockquote></details>

Done.
___
*[src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt, line 291 at r2](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57#-M_RV1Ty-kIYRqau2CET:-M_Rn7Cs6tN24IK0Ltpp:b-896fix) ([raw file](https://github.com/world-federation-of-advertisers/cross-media-measurement/blob/aba85d142f239932d8f933bdd4335a12ce2718ad/src/main/kotlin/org/wfanet/measurement/loadtest/CorrectnessImpl.kt#L291)):*
<details><summary><i>Previously, kungfucraig (Craig Wright) wrote…</i></summary><blockquote>

Raise and exception for the else clauses?
</blockquote></details>

Done.

**Review by kungfucraig — APPROVED** [review]
> Reviewed 4 of 5 files at r1, 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/world-federation-of-advertisers/cross-media-measurement/57)* status: all files reviewed, 5 unresolved discussions (waiting on @kungfucraig and @wangyaopw)

**Pull request merged**